### PR TITLE
Type Promotion for complex and bool pair

### LINKED ISSUE #36057: Runtime Error adding bool and complex tensors
Script to reproduce: 

```
import torch
a = torch.tensor((True,), dtype=torch.bool)
b = torch.tensor((1 + 1j,), dtype=torch.complex128)
print(a + b)
```

Result:

`RuntimeError: dtype != ScalarType::Undefined INTERNAL ASSERT FAILED at ../aten/src/ATen/native/TensorIterator.cpp:124, please report a bug to PyTorch.`

cc @ezyang @anjali411 @dylanbespalko

### PR DESCRIPTION
**Pull request by anjali411**
Fixes https://github.com/pytorch/pytorch/issues/36057

### CODE REVIEW

File: test/test_complex.py [hunk 1]
```diff
@@ -33,5 +33,13 @@ def test_copy_real_imag_methods(self):
         self.assertEqual(complex_tensor.copy_real(), real)
         self.assertEqual(complex_tensor.copy_imag(), imag)
 
+    # issue: https://github.com/pytorch/pytorch/issues/36057
+    def test_bool_complex_add(self):
```
Comment (nairbv) [inline test/test_complex.py:37]: it seems like this should just be another case or dtype entry in an existing test from test_type_promotion, this seems like a duplicate test

File: test/test_type_promotion.py [hunk 2]
```diff
@@ -118,6 +118,10 @@ def test_complex_scalar_mult_tensor_promotion(self, device):
         b = torch.tensor([2j, 2j], device=device)
         self.assertEqual(a, b)
         self.assertEqual(a.dtype, b.dtype)
+        a = a + torch.tensor([True, ])
+        b = b + 1
+        self.assertEqual(a, b)
+        self.assertEqual(a.dtype, b.dtype)
```
Comment (nairbv) [inline test/test_type_promotion.py:124]: nit: you can also add an `exact_dtype` arg to assertEquals to check the dtype (this was added recently I think).

**Comment by nairbv** [comment]
It might also be helpful to have some test checking the complex equivalent of this behavior:
```
> torch.tensor([1.]) & torch.tensor([True])
RuntimeError: "bitwise_and_cpu" not implemented for 'Float'
```
Right now with complex that gives the "internal assert failed" error. After this change it won't, but we won't want to allow bitwise ops on complex/bool combos, so we should verify that we're still disallowing that combination for this kind of op.

**Comment by anjali411** [comment]
fixed in #36747

**Pull request closed without merging**